### PR TITLE
Use pnpm instead of npm

### PR DESCRIPTION
**Pull request by schroedan**
The package manager pnpm is faster than npm and supports patches.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,7 +74,5 @@
   "nx": {
     "includedScripts": []
   },
-  "workspaces": [
-    "nodes/*"
-  ]
+  "packageManager": "pnpm@9.4.0"
```
Comment (MaSpeng) [inline package.json:77]: What is the benefit of providing this information? Is it necessary to pin it to this specific version, would it not be sufficient to pin it only to the major version?

File: .github/workflows/n8n-dependency-update.yaml [hunk 2]
```diff
@@ -59,7 +64,7 @@ jobs:
 
       - name: Update @types/node
         run: |
-          npm install @types/node@~${{ steps.target-versions.outputs.node_types_version }} \
+          pnpm add @types/node@~${{ steps.target-versions.outputs.node_types_version }} \
```
Comment (MaSpeng) [inline .github/workflows/n8n-dependency-update.yaml:67]: I think you need to add the option `--ignore-workspace-root-check`, otherwise pnpm will fail when you execute this command in the top-level folder.

See https://pnpm.io/cli/add#--ignore-workspace-root-check

This would also affect all other `add` commands as well.

File: .github/workflows/n8n-dependency-update.yaml [hunk 3]
```diff
@@ -21,22 +22,26 @@ jobs:
...
-        run: npm ci
+        run: pnpm install --frozen-lockfile
 
       - name: Retrieve target versions
         id: target-versions
         # yamllint disable rule:line-length
         run: |
           set -e
-          n8n_version=$(npm list n8n --json | jq --raw-output --exit-status '.dependencies.n8n.version')
+          n8n_version=$(pnpm list n8n --filter @skriptfabrik/n8n-nodes --json | jq --raw-output --exit-status '.[0].devDependencies.n8n.version')
```
Comment (MaSpeng) [inline .github/workflows/n8n-dependency-update.yaml:42]: What's the benefit of this filter, and why did you not use it in the other places as well?

File: nodes/barcode/.eslintignore [hunk 4]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (MaSpeng) [inline nodes/barcode/.eslintignore:1]: Why did you add this additional `.eslintignore`?

The root `.eslintignore` already contains node_modules, and at least for me, I never encountered a node_modules folder inside a package folder.

File: .gitignore [hunk 5]
```diff
@@ -11,7 +11,7 @@
 /coverage
 
 # Dependencies
-/node_modules
+node_modules
```
Comment (MaSpeng) [inline .gitignore:14]: How did it happen to you that there were multiple node_modules folders present?

**Review by MaSpeng — CHANGES_REQUESTED** [review]

File: .github/workflows/create-github-release.yaml [hunk 6]
```diff
@@ -53,17 +53,20 @@ jobs:
...
       - name: Setup Node.js
         uses: actions/setup-node@v4
         with:
-          cache: npm
+          cache: pnpm
           node-version: ${{ env.NODE_VERSION }}
 
       - name: Install dependencies
-        run: npm ci
+        run: pnpm install --frozen-lockfile
```
Comment (MaSpeng) [inline .github/workflows/create-github-release.yaml:66]: I discovered that `--frozen-lockfile` defaults to true in a CI environment.

https://pnpm.io/cli/install#--frozen-lockfile

File: .github/workflows/nx-dependency-update.yaml [hunk 7]
```diff
@@ -25,21 +28,25 @@ jobs:
...
 
       - name: Migrate nx
         run: |
-          npx nx migrate ${{ steps.dependabot-metadata.outputs.new-version }} \
+          pnpm exec nx migrate ${{ steps.dependabot-metadata.outputs.new-version }} \
             --from nx@${{ steps.dependabot-metadata.outputs.previous-version }}
 
       - name: Install updated depenencies
-        run: npm install
+        run: pnpm install
```
Comment (MaSpeng) [inline .github/workflows/nx-dependency-update.yaml:49]: Please add `--frozen-lockfile false`, otherwise this will fail in the CI.

See https://pnpm.io/cli/install#--frozen-lockfile

**Review by MaSpeng — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (schroedan) [reply package.json:77]: The reason: https://github.com/pnpm/action-setup/tree/v4/?tab=readme-ov-file#version

But we can also just pin it to the major version.

File: .github/workflows/n8n-dependency-update.yaml (hunk 3 above)
Comment (schroedan) [reply .github/workflows/n8n-dependency-update.yaml:42]: The command `pnpm list n8n` lists all installations of `n8n` in the entire workspace. The result is an array of objects whose name is the name of the package. We only want to get the version of n8n in the root package.

File: nodes/barcode/.eslintignore (hunk 4 above)
Comment (schroedan) [reply nodes/barcode/.eslintignore:1]: The target `lint` seems to start `eslint .` in the directory of each node directory. So the root `.eslintignore` is out of scope.

File: .gitignore (hunk 5 above)
Comment (schroedan) [reply .gitignore:14]: The command `pnpm install` will install all `node_modules` of all packages in the workspace.

File: package.json (hunk 1 above)
Comment (MaSpeng) [reply package.json:77]: Nice catch, the documentation seems to be incorrect, the action works flawlessly in the test repository without providing this information to the action or the package.json. 

But sure we should follow the documentation and I would say we pin it at least to the major version.

File: .github/workflows/n8n-dependency-update.yaml (hunk 3 above)
Comment (MaSpeng) [reply .github/workflows/n8n-dependency-update.yaml:42]: Ah thanks, that's better you are right, thanks for the explanation, did not recognize it myself.

File: .gitignore (hunk 5 above)
Comment (MaSpeng) [reply .gitignore:14]: Ah right, I see I made a configuration wich prevented this from happening but could be invalid.

File: package.json (hunk 1 above)
Comment (MaSpeng) [reply package.json:77]: That's bad, I hope they will improve the documentation from the action so it reflects this requirement. The Node.js documentation is not very detailed about this property due to its experimental state.

**Review by MaSpeng — APPROVED** [review]
